packages/snaps-controllers/src/interface/SnapInterfaceController.ts

```diff
@@ -440,6 +469,41 @@ export class SnapInterfaceController extends BaseController<
     );
   }
 
+  /**
+   * Get the selected account in the client.
+   *
+   * @returns The selected account.
+   */
+  #getSelectedAccount() {
+    return this.messagingSystem.call(
+      'AccountsController:getSelectedMultichainAccount',
+    );
+  }
+
+  /**
+   * Get a list of accounts for the given chain IDs.
+   *
+   * @param chainIds - The chain IDs to get the accounts for.
+   * @returns The list of accounts.
+   */
+  #listAccounts(chainIds?: CaipChainId[]) {
+    if (!chainIds || chainIds.length === 0) {
+      return this.messagingSystem.call(
+        'AccountsController:listMultichainAccounts',
+      );
+    }
+
+    const accounts = chainIds.flatMap((chainId) =>
+      this.messagingSystem.call(
+        'AccountsController:listMultichainAccounts',
+        chainId,
+      ),
+    );
+
+    // dedupe accounts that could have multiple scopes in common
+    return Array.from(new Set(accounts));
+  }
```

Comment:
Sorry, now that I think about it, it is probably a lot faster to have one call to `listMultichainAccounts` and then filter out any scopes we don't need? Then we don't need to dedupe and flatten, WDYT?

Reply:
Yup that's true that it would probably be faster and cheaper to do it that way